proposals/lip-0051.md

```diff
@@ -814,484 +2596,147 @@ genesisTokenStoreSchema = {
+                        }
+                    }
+                }            
+                
             }
         }
     }
 }
 ```
 
+  
+
 #### Genesis State Initialization
 
 During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
 
 Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the Token module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisTokenStoreSchema` schema, given above.
 
 * Initial checks on the properties of `genesisBlockAssetObject`:
-  * Across all elements of the `userSubstore` array, all `address` values must have length `ADDRESS_LENGTH`.
   * Across all elements of the `userSubstore` array, the pairs `address, tokenID` must be unique (each pair appears at most once, though a given address can appear multiple times with different token IDs).
-  * The `userSubstore` array must be in lexicographical order of `address`. For a given `address`, the entries must be in ascending order of `tokenID.chainID`. For a given `address` and `tokenID.chainID`, the entries must be in ascending order of `tokenID.localID`.
+  * The `userSubstore` array must be in lexicographical order of `address`. For a given `address`, the entries must be in ascending order of `tokenID`.
   * For each `address, tokenID` pair:
-    * The values used as `moduleID` in the `lockedBalances` array must be unique.
-    * The `lockedBalances` array must be in asscending order of `moduleID`.
+    * The values used as `module` in the `lockedBalances` array must be unique.
+    * The `lockedBalances` array must be in lexicographic order of `module` name.
     * All elements of the `lockedBalances` array must have `amount != 0`.
     * Either `availableBalance != 0` or `lockedBalances != []` must be true, i.e.: the entry must contain some tokens.
-  * Across all element of the `supplySubstore` array, all values given for `localID` must be unique.
-  * The `supplySubstore` array must be in ascending order of `localID`.
-  * Across all elements of the `escrowSubstore` array, the pairs `escrowChainID, localID` must be unique (each pair appears at most once).
-  * The `escrowSubstore` array must be in ascending order of `escrowChainID`. For a given `escrowChainID`, the entries must be in ascending order of `localID`.
-  * In the `terminatedEscrowSubstore` array, each element must be unique (appears at most once).
-  * The `terminatedEscrowSubstore` array must be in ascending order.
+  * Across all element of the `supplySubstore` array, all values given for `tokenID` must be unique.
+  * The `supplySubstore` array must be in ascending order of `tokenID`.
+  * Across all elements of the `escrowSubstore` array, the `tokenID` must be unique (it appears at most once).
+  * The `escrowSubstore` array must be in ascending order of `tokenID`. 
```

Comment:
missing validation for supportedTokenIDs store?

Reply:
Good catch. Added.